jest.config.cjs

```diff
@@ -1,4 +1,16 @@
 module.exports = {
-    preset: 'ts-jest',
-    testEnvironment: "jsdom",
-};
+  transform: {
+    '^.+\\.(t|j)sx?$': '@swc/jest',
+  },
+  testEnvironment: 'jsdom',
+  testEnvironmentOptions: {
+    customExportConditions: [''],
+  },
+  globals: {
```

Comment:
Yet another Jest shenanigan: it strips perfectly valid and present Node.js globals, breaking your test environment. You have to manually re-map all fetch-related globals here just so Jest would recognize them. 

Once again, not an issue with Vitest.